client/Wayland/wlfreerdp.c

```diff
@@ -346,7 +345,9 @@ static int wlfreerdp_run(freerdp* instance)
 		//if (WaitForMultipleObjects(count, &handles[1], FALSE, INFINITE)) {
 		if (freerdp_check_event_handles(instance->context) != TRUE)
 		{
-			printf("Failed to check FreeRDP file descriptor\n");
+			if (freerdp_get_last_error(instance->context) == FREERDP_ERROR_SUCCESS)
```

Comment:
Again should be !=

Reply:
No, only print something if no error has been set (which already printed a more detailed error message)